doc/getting-started/generating-transactions.md

```diff
@@ -1,14 +1,105 @@
+
+## Example code
+
+The example below builds a transaction with all 2 of the 3 input types: key and bootstrap.
+Multisig (script) inputs are essentially identical to key inputs, but using the scripthash instead of the keyhash, however they are not supported for implicit fee calculation yet.
+Fees are automatically calculated and sent to a change address in the example.
+
+
+```javascript
+// instantiate the tx builder with the Cardano protocol parameters - these may change later on
+const txBuilder = CardanoWasm.TransactionBuilder.new(
+    // all of these are taken from the mainnet genesis settings
+    CardanoWasm.LinearFee.new(CardanoWasm.BigNum.from_str('44'), CardanoWasm.BigNum.from_str('155381')),
+    CardanoWasm.BigNum.from_str('1000000'),
+    CardanoWasm.BigNum.from_str('500000000'),
+    CardanoWasm.BigNum.from_str('2000000')
+);
+
+// add a keyhash input - for ADA held in a Shelley-era normal address (Base, Enterprise, Pointer, Reward)
+const prvKey = CardanoWasm.PrivateKey.from_bech32("ed25519e_sk16rl5fqqf4mg27syjzjrq8h3vq44jnnv52mvyzdttldszjj7a64xtmjwgjtfy25lu0xmv40306lj9pcqpa6slry9eh3mtlqvfjz93vuq0grl80");
+txBuilder.add_key_input(
+    prvKey.to_public().hash(),
+    CardanoWasm.TransactionHash.from_bytes(
+    Buffer.from("8561258e210352fba2ac0488afed67b3427a27ccf1d41ec030c98a8199bc22ec", "hex"),
+    ), // tx hash
+    3, // index
+);
+
+// add a bootstrap input - for ADA held in a Byron-era address
+const byronAddress = CardanoWasm.ByronAddress.from_base58("Ae2tdPwUPEZLs4HtbuNey7tK4hTKrwNwYtGqp7bDfCy2WdR3P6735W5Yfpe");
+txBuilder.add_bootstrap_input(
+    byronAddress,
+    CardanoWasm.TransactionInput.new(
+    CardanoWasm.TransactionHash.from_bytes(
+        Buffer.from("488afed67b342d41ec08561258e210352fba2ac030c98a8199bc22ec7a27ccf1", "hex"),
+    ), // tx hash
+    0, // index
+    ),
+    CardanoWasm.BigNum.from_str('3000000')
+);
+
+// base address
+const shelleyOutputAddress = CardanoWasm.Address.from_bech32("addr_test1qpu5vlrf4xkxv2qpwngf6cjhtw542ayty80v8dyr49rf5ewvxwdrt70qlcpeeagscasafhffqsxy36t90ldv06wqrk2qum8x5w");
+// pointer address
+const shelleyChangeAddress = CardanoWasm.Address.from_bech32("addr_test1gz2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzerspqgpsqe70et");
+
+// add output to the tx
+txBuilder.add_output(
+    CardanoWasm.TransactionOutput.new(
+    shelleyOutputAddress,
+    CardanoWasm.BigNum.from_str("1000000"),
+    ),
+);
+
+// set the time to live - the absolute slot value before the tx becomes invalid
+txBuilder.set_ttl(410021);
+
+// calculate the min fee required and send any change to an address
+txBuilder.add_change_if_needed(shelleyChangeAddress)
+
+// once the transaction is ready, we build it to get the tx body without witnesses
+const txBody = txBuilder.build();
+const txHash = CardanoWasm.hash_transaction(txBody);
+const witnesses = CardanoWasm.TransactionWitnessSet.new();
+
+// add keyhash witnesses
+const vkeyWitnesses = CardanoWasm.VkeyWitnesses.new();
```

Comment:
We should maybe link to the spec where it mentions what witnesses are needed for which thing, since it's not obvious for example that a stake key registration & delegation requires a VkeyWitness for the staking key, but the stake key registration certificate doesn't require a witness.
Also not obvious that registration & deregistration includes a deposit & refund